### ISSUE TITLE
Not working with support.mozilla.org

### ISSUE THREAD

**Issue by CNG** [0x6b/copy-selection-as-markdown#57]
Hello, is there some kind of special case or known issue preventing this from working with support.mozilla.org? If not, here's a bug report.

---

With the default preferences as well as with **Log debug message in the console** enabled, I am unable to copy selection as Markdown for at least these two web pages:

* https://support.mozilla.org/en-US/kb/setting-certificate-authorities-firefox
* https://support.mozilla.org/en-US/

For example, with **Log debug message in the console** enabled:

1. Open _Developer Tools_ > _Console_.
2. Go to [Setting Up Certificate Authorities (CAs) in Firefox](https://support.mozilla.org/en-US/kb/setting-certificate-authorities-firefox).
3. Select the title, "Setting Up Certificate Authorities (CAs) in Firefox".
4. Right click and select _Copy_. (If you omit this step, the pasted text will still be whatever was on the clipboard. This is just to make the actual/expected explicit.)
5. Select the first sentence of the first bullet, "Setting the _ImportEnterpriseRoots_ key to **true** will cause Firefox to trust root certificates.".
6. Right click and select _Copy Selection as Markdown_.
7. In a text box, paste the clipboard.

**Actual Result**

Pasted text is the plain copied text:

```
Setting Up Certificate Authorities (CAs) in Firefox
```

Console shows no output.

**Expected Result**

Pasted text is the Markdown version of copied text with the reference:

```
[Setting Up Certificate Authorities (CAs) in Firefox | Firefox for Enterprise Help](file:///home/cgorichanaz/Sorting/Setting%20Up%20Certificate%20Authorities%20(CAs)%20in%20Firefox%20%7C%20Firefox%20for%20Enterprise%20Help.html)

> Setting the _ImportEnterpriseRoots_ key to **true** will cause Firefox to trust root certificates.
```

Console shows output like this:

```
/* --- copy-selection-as-markdown debug information --- */ copy.js:56:14
/* --- INPUT ------------------------------------------ */ copy.js:57:14
Setting the <em>ImportEnterpriseRoots</em> key to <strong>true</strong>
 will cause Firefox to trust root certificates. copy.js:58:14
/* --- OUTPUT------------------------------------------ */ copy.js:59:14
> Setting the _ImportEnterpriseRoots_ key to **true** will cause Firefox to trust root certificates. copy.js:60:14
/* --- URL -------------------------------------------- */ copy.js:61:14
https://support.mozilla.org/en-US/kb/setting-certificate-authorities-firefox copy.js:62:14
/* ---------------------------------------------------- */
```

**Notes**

I generated the above console output by opening https://support.mozilla.org/en-US/kb/setting-certificate-authorities-firefox and saving the web page locally, then visiting that local copy and doing the above procedure to copy as Markdown. Then I replaced `file:///home/cgorichanaz/Setting%20Up%20Certificate%20Authorities%20(CAs)%20in%20Firefox%20%7C%20Firefox%20for%20Enterprise%20Help.html` with `https://support.mozilla.org/en-US/kb/setting-certificate-authorities-firefox` in the console output.

Environment:
* 75.0 (64-bit) Mozilla Firefox for Arch Linux (package `firefox 75.0-1`)
* OS: Arch Linux, kernel 5.6.5-arch3-1

**Comment by 0x6b** [comment]
Hi @CNG, thanks for taking your time to open an issue. Yes, it's intended behavior as, by default, add-ons cannot interfere some pre-configured domains for security reason. You can remove `support.mozilla.org` from the `extensions.webextensions.restrictedDomains` preference at `about:config`. Please try on your end as changing it worked for me.

See [Configuration Editor for Firefox | Firefox Help](https://support.mozilla.org/en-US/kb/about-config-editor-firefox) for how to change the preference.

**Comment by CNG** [comment]
tldr; After some more testing, it's working for me now, so thank you!

Apologies for the slow reply here.
* First, wanted to thank you for the quick reply!
* Second, I couldn't get this to work and wanted to test more. I had removed all the entries from my `extensions.webextensions.restrictedDomains` and it still was not working. I restarted Firefox a few times. Then, I went to this plugin's settings and toggled the logging setting and clicked *Save*, and then it started working. I'm not totally sure (It's possible I had something strange in my setup.) but wanted to mention here in case someone runs into this.
